modules/database/src/adapters/mongoose-adapter/index.ts

```diff
@@ -87,26 +88,24 @@ export class MongooseAdapter extends DatabaseAdapter<MongooseSchema> {
     //@ts-ignore
     newSchema.collectionName = viewName;
 
-    try {
-      const viewModel = new MongooseSchema(
-        this.grpcSdk,
-        this.mongoose,
-        newSchema as ConduitSchema,
-        model.originalSchema,
-        this,
-        true,
-        query,
-      );
-      await viewModel.model.createCollection({
-        viewOn: model.originalSchema.collectionName,
-        pipeline: JSON.parse(query.mongoQuery),
-      });
-      this.views[viewName] = viewModel;
-    } catch (e: any) {
-      if (!e.message.includes('Cannot overwrite')) {
-        throw e;
-      }
+    if (existingView && !isQueryEqual) {
```

Comment:
you also need to delete the this.views[viewName] reference so that the db doesn't thing a view exists which might not exist anymore. The case is that the old view is deleted, but the new one has a query error and is not constructed successfully

Reply:
Already done in deleteView()